DCQL: express desired credential issuers (#393)
openid/OpenID4VP: openid-4-verifiable-presentations-1_0.md

no specific constraints are placed on the metadata or validity of the requested
Credential.

`trusted_authorities`:
: OPTIONAL. A non-empty array of objects as defined in (#dcql_trusted_authorities) that

> specifies expected issuers or trust frameworks that an issuer belongs to, that the

Reviewer: ```suggestion
specifies expected authorities or trust frameworks that certify issuers, that the
```

> `accepted_issuers`:
> : OPTIONAL. A non-empty array of objects as defined in (#dcql_accepted_issuers) that
> specifies expected issuers or trust frameworks that an issuer belongs to, that the
> Verifier will accept. Every Credential returned by the Wallet SHOULD match at least
> one of the conditions present in the corresponding `accepted_issuers` array if present.
> 
> Note that Relying Parties must verify that the issuer of a received presentation is
> trusted on their own and this feature mainly aims to help data minimization by not
> revealing information that would likely be rejected.
> 

Reviewer 2: ```suggestion
`accepted_issuers`:
: OPTIONAL. A non-empty array of objects as defined in (#dcql_accepted_issuers) that
specifies expected Credential Issuers that the
Verifier accepts. Every Credential returned by the Wallet SHOULD match at least
one of the conditions present in the corresponding `accepted_issuers` array if present.
Relying Parties using this parameter MUST ensure that the Credential Issuers specified are both active and trusted at the time of the query.
```

I would not mix trust frameworks identifiers with credential identifiers in the same parameter

no specific constraints are placed on the metadata or validity of the requested
Credential.

`trusted_authorities`:

> : OPTIONAL. A non-empty array of objects as defined in (#dcql_trusted_authorities) that

Reviewer 2: In allegato this pagamento agraph, issuer -> Issuer since It Is intended to be credential issuers

Author: Fixed in latest commit

### Accepted Issuer Query {#dcql_accepted_issuers}

An Accepted Issuer Query is an object representing information that helps to identify an issuer
or the trust framework that an issuer is belonging to. A Credential is identified as a match
to an Accepted Issuer Query if it to matches with one of the provided values in one of the provided
types. How exactly the matching works is defined for the different types below

Note that explicit issuer matching can also be achieved using claim value matching for certain
Credential formats (e.g., value matching the `iss` claim in an SD-JWT). The methods provided in
accepted_issuers are more tailored towards matching trust frameworks and those that require more
complex matching logic (e.g., certificate thumbprints).

Each entry in `accepted_issuers` MUST be an object with the following properties:

`type`:
: REQUIRED. A string uniquely identifying the type of information about the issuer trust framework.
Possible types are defined below.

`values`:
: REQUIRED. An array of strings, where each string (value) contains information specific to the

> used Accepted Issuer Query type that allows to identify an issuer or a trust ecosystem that an

Reviewer 3: ```suggestion
used Accepted Issuer Query type that allows to identify an issuer, trust framework, or a federation that an
```

certificate in the trust chain of a matching Credential.

Below is a non-normative example of such an entry of type `aki`:
```json
{
  "type": "aki",
  "values": ["TODO"]
}
```

#### Verified Issuer Certificate Authority List (VICAL)

Type:
: `"vical"`

Value:
: The identifier (can also be a URL) of a Verified Issuer Certificate Authority List (VICAL) as specified in
[@ISO.18013-5]. The VICAL contains a list of Issuing Authority Certificate Authorities (IACA), that provide
DER encoded X.509 Certificates and corresponding allowed document types (an array of DocType identifiers).
This Type MUST only be used in combination with Credentials of the Credential Format mdoc. The trust chain
of a matching Credential MUST contain at last one X.509 Certificate entry in the IACAs and the allowed
DocTypes of the corresponding IACA entry MUST match with the doctype of the Credential.

Below is a non-normative example of such an entry of type `vical`:
```json
{
  "type": "vical",
  "values": ["https://vical.example.com"]
}
```

#### ETSI Trusted List

Type:
: `"etsi_tl"`

Value:
: The identifier (can also be a URL) of a Trusted List as specified in ETSI TS 119 612.

> TODO: How to express restrictions within that TL? e.g., filtering by service type?

Author: My biggest question here is if the expectation is for a specific trusted list (of lists) to only include one type (e.g., all PID issuers), or if we will have bigger lists where we need to filter within that one list for different types.

I started reading ARF annex 2 for PID issuance and currently my understanding would be that for the PID we have a special list of PID Providers (so no filtering necessary). Is that correct and are we fine to start with the assumption that no further filtering is necessary here?

@babisRoutis are you be any chance able to provide a bit more information on this?

Reviewer 3: Good point. It would be nice to make it more universal.

Below is a non-normative example of such an entry of type  `etsi_tl`:
```json
{
  "type": "etsi_tl",
  "values": ["https://ec.europa.eu/tools/lotl/eu-lotl.xml"]
}
```

#### OpenID Federation

Type:
: `"oidf"`

Value:
: The `Entity Identifier` as defined in Section 1 of [@!OpenID.Federation] that is bound to

> an entity in an OpenID Federation System. While this Entity Identifier could be any entity in

Reviewer 3: We could change OpenID Federation System to a federation or an ecosystem or something along these lines.

that ecosystem, this entity would usually have the Entity Configuration of a Trust Anchor.

> The trust chain of a matching credential MUST contain the given Entity Identifier.

Reviewer 3: ```suggestion
The trust chain of a matching credential MUST contain the Entity Configuration issued by the Entity identified by the given Entity Identifier.
```

Reviewer 3: There is no direct use case for that but logically we should also consider including option to use intermediary ids. Any thoughts?

Author: Sounds reasonable with my limited knowledge of OIDF, @selfissued, @peppelinux ?

Reviewer 4: With the disclaimer that I'm no expert in OIDF, I believe there is no such thing as "the trust chain" for a given credential. There could be multiple trust roots that a credential could chain up to (especially assuming that roots will cross-certify each other as trusted peers, e.g. in the EU). So I would recommend something like "A valid trust path, terminating at the given Entity Identifier, must be constructible from a matching credential".

Author: Agreed that trust chain doesn't seem to be the best wording here. I am a bit hesitant with the word "terminating" as it sounds like we want to allow intermediate entities and terminating could be interpreted as reaching a Trust Anchor?

Would this work for you @lj-raidiam , @balfanz 

```suggestion
A valid trust path, reaching the given Entity Identifier, must be constructible from a matching credential.
```

Reviewer 5: I'm curious how the processing of this would work. Knowing if you can build a chain including the requested entity id often involves a (complicated) resolving process. As @balfanz pointed out, multiple paths could exist, so you'd either have to:
- store (a subset of) the possible trust chains with the credential, so you can perform this offline 
- resolve the trust chain dynamically to match (this doesn't seem like a good approach, credential matching should be performed locally IMO).

Reviewer 5: Reaching would still suggest an end, maybe including works?

```suggestion
A valid trust path, including the given Entity Identifier, must be constructible from a matching credential.
```

Reviewer 3: > I'm curious how the processing of this would work. Knowing if you can build a chain including the requested entity id often involves a (complicated) resolving process. As @balfanz pointed out, multiple paths could exist, so you'd either have to:
> 
> * store (a subset of) the possible trust chains with the credential, so you can perform this offline
> * resolve the trust chain dynamically to match (this doesn't seem like a good approach, credential matching should be performed locally IMO).

Yes, wallets will have to store it for all credentials to do the filtering. Perhaps in more complex case (like two paths) this filtering will not be ideal. Verifiers will have a harder work to do as they will have to resolve it dynamically in most cases.

> TODO: Do we need to specify more like Entity Type for matching?

Reviewer 3: If OpenID Federation 4 Wallet Architectures is supported the issuer could be only https://openid.net/specs/openid-federation-wallet-1_0.html#section-6.2 
@peppelinux any thoughts?

Note that multiple Credential Queries in a request MAY request a presentation of the same Credential.

### Accepted Issuer Query {#dcql_accepted_issuers}

An Accepted Issuer Query is an object representing information that helps to identify an issuer
or the trust framework that an issuer is belonging to. A Credential is identified as a match
to an Accepted Issuer Query if it to matches with one of the provided values in one of the provided
types. How exactly the matching works is defined for the different types below

Note that explicit issuer matching can also be achieved using claim value matching for certain
Credential formats (e.g., value matching the `iss` claim in an SD-JWT). The methods provided in
accepted_issuers are more tailored towards matching trust frameworks and those that require more
complex matching logic (e.g., certificate thumbprints).

Each entry in `accepted_issuers` MUST be an object with the following properties:

`type`:

> : REQUIRED. A string uniquely identifying the type of information about the issuer trust framework.
> Possible types are defined below.

Reviewer 5: "Possible types" suggests this is not extendable. Should this say "defined by this specification" as is done with client_id_scheme?

```suggestion
: REQUIRED. A string uniquely identifying the type of information about the issuer trust framework.
Types defined by this specification are defined below.
```

`values`:
: REQUIRED. An array of strings, where each string (value) contains information specific to the
used Accepted Issuer Query type that allows to identify an issuer, trust framework, or a federation that an
issuer belongs to.

Below are descriptions for the different Type Identifiers (string), the description on how to interpret
and perform the matching logic for each provided value.

> #### X.509 Thumbprint

Reviewer 6: Reasoning to introduce this was for the CA that do not have AKIs. (How many of certs don't have AKIs?)

> #### X.509 Thumbprint
> 
> Type:
> : `"x5t#S256"`
> 
> Value:
> : The base64url encoded SHA-256 digest (a.k.a. thumbprint, fingerprint) of the DER encoding of an X.509
> certificate as defined in Section 4.1.8 of [@!RFC7515]. This thumbprint MUST match with the thumbprint
> of an X.509 certificate in the trust chain of the Credential.
> 
> Below is a non-normative example of such an entry of type `x5t#S256`:
> ```json
> {
>   "type": "x5t#S256",
>   "values": ["455943cf819425761d1f950263ebf54755d8d684c25535943976f488bc79d23b"]
> }
> ```
> 

Reviewer 7: I'd suggest removing X.509 thumbprint because it is less useful than `aki` and doesn't have any benefit over `aki`. It has actually a couple of downsides like having issues with re-issued certificates.
```suggestion
```

Author: The 2 biggest reasons why I chose to include it in the initial proposal was 
- x5t is somewhat commonly used in the oauth world -> keep the choices consistent
- AuthoriyKeyIdentifier is an extension and I was not sure if every ecosystem that would want to use AKI.

That said, I am happy to remove it if everyone can use / is happy with AKI. Less choices is always a good thing. It would also be easy to re-introduce x5t later on.

> #### Authority Key Identifier

Reviewer 6: ISO feedback. less options the better. AKI alone is probably sufficient. AKI meets ISO requirement.

Reviewer 8: Then we should consider dropping VICAL.

Type:
: `"aki"`

Value:
: The base64url encoded Authority Key Identifier element as defined in Section 4.2.1.1 of [@!RFC5280].
The raw byte representation of this element MUST match with the AuthorityKeyIdentifier element of an X.509
certificate in the trust chain of a matching Credential.

Below is a non-normative example of such an entry of type `aki`:
```json
{
  "type": "aki",
  "values": ["s9tIpPmhxdiuNkHMEWNpYim8S8Y"]
}
```

> #### Verified Issuer Certificate Authority List (VICAL)

Reviewer 6: DCP WG discussion. leaning to keep it in OpenID4VP because value is not defined as a URL, but an identifier that does not have to be resovled

Type:
: `"vical"`

Value:
: The identifier (can also be a URL) of a Verified Issuer Certificate Authority List (VICAL) as specified in
[@ISO.18013-5]. The VICAL contains a list of Issuing Authority Certificate Authorities (IACA), that provide
DER encoded X.509 Certificates and corresponding allowed document types (an array of DocType identifiers).
This Type MUST only be used in combination with Credentials of the Credential Format mdoc. The trust chain
of a matching Credential MUST contain at last one X.509 Certificate entry in the IACAs and the allowed
DocTypes of the corresponding IACA entry MUST match with the doctype of the Credential.

Below is a non-normative example of such an entry of type `vical`:
```json
{
  "type": "vical",
  "values": ["https://vical.example.com"]
}
```

#### ETSI Trusted List

Type:
: `"etsi_tl"`

Value:
: The identifier (can also be a URL) of a Trusted List as specified in ETSI TS 119 612 [@ETSI.TL].An ETSI

> Trusted List contains references to other Trusted Lists, creating a list of trustedf lists, or entries

Reviewer 5: ```suggestion
Trusted List contains references to other Trusted Lists, creating a list of trusted lists, or entries
```

for Trust Service Providers with corresponding service description and X.509 Certificates. The trust chain
of a matching Credential MUST contain at least one X.509 Certificate that matches one of the entires of the
Trusted List or its cascading Trusted Lists.

> Below is a non-normative example of such an entry of type  `etsi_tl`:

Reviewer 5: ```suggestion
Below is a non-normative example of such an entry of type `etsi_tl`:
```

```json
{
  "type": "etsi_tl",
  "values": ["https://lotl.example.com"]
}

> ```

Author: I might be missing something but it seems to me that how exactly the PID Issuer trusted list (or list of lists) will look like is currently not 100% clear- In that case I would propose to leave the ETSI TL as is (without more complex matching logic) and open an issue to revisit before going final

Reviewer 8: I agree. The current revision is a good starting point. Implementation experience will show what is needed on top.

#### OpenID Federation

Type:

> : `"oidf"`

Reviewer 9: ```suggestion
: `"openid_fed"`
```

At the risk of being a broken record, for branding reasons, we should always be spelling "OpenID" using all of its letters, and not shortening it to "OID", which means Object IDentifier in most identity contexts.  Thanks.

Value:
: The `Entity Identifier` as defined in Section 1 of [@!OpenID.Federation] that is bound to
an entity in an OpenID Federation System. While this Entity Identifier could be any entity in
that ecosystem, this entity would usually have the Entity Configuration of a Trust Anchor.
The trust chain of a matching credential MUST contain the given Entity Identifier.

> Below is a non-normative example of such an entry of type `oidf`: 

Author: ```suggestion
Below is a non-normative example of such an entry of type `openid_fed`: 
```

```json
{

>   "type": "oidf",

Reviewer 9: ```suggestion
  "type": "openid_fed",
```

Note that multiple Credential Queries in a request MAY request a presentation of the same Credential.

### Accepted Issuer Query {#dcql_accepted_issuers}

> An Accepted Issuer Query is an object representing information that helps to identify an issuer
> or the trust framework that an issuer is belonging to. A Credential is identified as a match
> to an Accepted Issuer Query if it to matches with one of the provided values in one of the provided
> types. How exactly the matching works is defined for the different types below
> 
> Note that explicit issuer matching can also be achieved using claim value matching for certain
> Credential formats (e.g., value matching the `iss` claim in an SD-JWT). The methods provided in
> accepted_issuers are more tailored towards matching trust frameworks and those that require more
> complex matching logic (e.g., certificate thumbprints).
> 
> Each entry in `accepted_issuers` MUST be an object with the following properties:
> 
> `type`:
> : REQUIRED. A string uniquely identifying the type of information about the issuer trust framework.
> Types defined by this specification are listed below.
> 
> `values`:
> : REQUIRED. An array of strings, where each string (value) contains information specific to the
> used Accepted Issuer Query type that allows to identify an issuer, trust framework, or a federation that an
> issuer belongs to.
> 
> Below are descriptions for the different Type Identifiers (string), the description on how to interpret
> and perform the matching logic for each provided value.

Reviewer 2: ```suggestion
An Accepted Issuer Query is an object representing information that identify one or more Credential Issuers. 
A Credential matches an Accepted Issuer Query if it aligns at least its Credential Issuer and the Credential Issuer configures the Credential as supported (eg: `iss` and `vct` for SD-JWT). Different Credential Types may use different matching approaches.

Each `accepted_issuers` object MUST have the following members:

`credential_issuer`:
: REQUIRED. A string uniquely identifying the Credential Issuer.

`trust_evaluations_supported`:
: REQUIRED. An array of strings, uniquely identifying the type of information about the trust evaluation mechanism the Credential Issuer MUST support.

`trust_frameworks_supported`:
: OPTIONAL. An array of strings, where each string (value) contains information specific to the
supported trust framework the Credential Issuer MUST belong to, to be considered a valid Credential Issuer.

Below are descriptions for the different Type Identifiers (string), the description on how to interpret
and perform the matching logic for each provided value.
```

Author: I don't think your proposal would work with the current scope of the PR:
- Part of the problem initially discussed  in the issue was that not every issuer is easily identified by a string (see the Authority Key Identifier discussion)
- The discussion in the issue basically spanned 2 problems:
  - Direct issuer matching -> This will be done by value matching unless impossible (AKI)
  - Expressing requirements about the trust chain/path -> Instead of directly identifying the issuer, expressing some expected properties/entities in the trust chain

This PR mainly covers the second part - describing some requirements that the trust chain/path of a credential has to fulfil. I think some of the confusion stems from the rather unfitting name "accepted_issuers"?

Note that multiple Credential Queries in a request MAY request a presentation of the same Credential.

### Trusted Authorities Query {#dcql_trusted_authorities}

> A Trusted Authorities Query is an object representing information that helps to identify an issuer
> or the trust framework that an issuer belongs to. A Credential is identified as a match

Reviewer: ```suggestion
A Trusted Authorities Query is an object representing information that helps to identify an authority
or the trust framework that certifies issuers. A Credential is identified as a match
```

> or the trust framework that an issuer is belonging to. A Credential is identified as a match

Reviewer 10: nit:belongs to

Author: thanks for catching that - fixed with last commit

> to a Trusted Authorities Query if it to matches with one of the provided values in one of the provided

Reviewer 10: nit: if it matches

types. How exactly the matching works is defined for the different types below

> Note that explicit issuer matching can also be achieved using claim value matching for certain

Reviewer: ```suggestion
Note, as opposed to authority or trust framework matching, explicit issuer matching should use claim value matching for certain
```

should use claim value matching if possible (e.g., value matching the `iss` claim in an SD-JWT).
The methods provided in `trusted_authorities` are for more indirect matching logic required by trust

> frameworks or X.509 PKI.

Reviewer 11: I don't understand this recommendation. 
For example, issuer matching based on the AKI of a certificate owned by a particular issuer can be very direct. In that example I think using AKI is the more reliable, more privacy and security preserving than a mechanism based on claim matching.

Author: This recommendation is not intended to say that some mechanisms are better or worse, but explain that if ecosystems want to do direct issuer matching based on claims, then `trusted_authorities` is not the right place and value-based matching should be used instead (e.g, matching `iss` of an SD-JWT to the value of an URL or a DID).

Reviewer 9: On the 20-Feb-25 working group call, there was support for discussing the privacy implications of online trust validation processes in Privacy Considerations text, and not blocking this PR on this.  I believe that @c2bo is going to create the PR.

> Note that explicit issuer matching can also be achieved using claim value matching for certain
> Credential formats (e.g., value matching the `iss` claim in an SD-JWT). The methods provided in
> `trusted_authorities` are tailored towards matching trust frameworks and more direct issuer matching
> variants that require more complex matching logic (e.g., Authority Key Identifier).

Reviewer: ```suggestion
Note direct issuer matching should use claim value matching for certain Credential formats (e.g., value matching the `iss` claim in an SD-JWT).  The methods provided in `trusted_authorities` are for more indirect methods using trust frameworks or variants that require authority matching.
```

> variants that require more complex matching logic (e.g., certificate thumbprints).

Reviewer 5: Certificate thumbprints are not part of this PR anymore. Would aki suffice as an example here? Or wouldn't that be called direct issuer matching?

Author: Changed the example to AKI

Each entry in `trusted_authorities` MUST be an object with the following properties:

`type`:
: REQUIRED. A string uniquely identifying the type of information about the issuer trust framework.
Types defined by this specification are listed below.

`values`:
: REQUIRED. An array of strings, where each string (value) contains information specific to the
used Trusted Authorities Query type that allows to identify an issuer, trust framework, or a federation that an
issuer belongs to.

Below are descriptions for the different Type Identifiers (string), the description on how to interpret
and perform the matching logic for each provided value.

#### Authority Key Identifier

Type:
: `"aki"`

Value:
: The base64url encoded Authority Key Identifier element as defined in Section 4.2.1.1 of [@!RFC5280].
The raw byte representation of this element MUST match with the AuthorityKeyIdentifier element of an X.509

> certificate in the trust chain of a matching Credential.

Reviewer 11: Can we be more specific than 'trust chain'? For example 'the AKI of one of the certificates carried in the chain carried present in the header parameters of the credential. Note that chain can consist of a single certificate.

Author: Would this work for you @martijnharing ?
```suggestion
certificate in the certificate chain present in the credential (e.g., in the header of an mdoc or SD-JWT).
Note that the chain can consist of a single certificate.
```

Below is a non-normative example of such an entry of type `aki`:
```json
{
  "type": "aki",
  "values": ["s9tIpPmhxdiuNkHMEWNpYim8S8Y"]
}
```

#### ETSI Trusted List

Type:
: `"etsi_tl"`

Value:

> : The identifier (can also be a URL) of a Trusted List as specified in ETSI TS 119 612 [@ETSI.TL].An ETSI

Reviewer 11: Since a URL is mentioned, can we add language that prohibits using the URL from retrieving the trust chain because of the privacy implications of such a fetch?

Author: I can remove URL here if that helps? 
```suggestion
: The identifier of a Trusted List as specified in ETSI TS 119 612 [@ETSI.TL].An ETSI
```

Trusted List contains references to other Trusted Lists, creating a list of trusted lists, or entries
for Trust Service Providers with corresponding service description and X.509 Certificates. The trust chain
of a matching Credential MUST contain at least one X.509 Certificate that matches one of the entries of the
Trusted List or its cascading Trusted Lists.

Below is a non-normative example of such an entry of type `etsi_tl`:
```json
{
  "type": "etsi_tl",
  "values": ["https://lotl.example.com"]
}
```

#### OpenID Federation

Type:
: `"openid_fed"`

Value:

> : The `Entity Identifier` as defined in Section 1 of [@!OpenID.Federation] that is bound to
> an entity in an OpenID Federation System. While this Entity Identifier could be any entity in

Author: ```suggestion
: The `Entity Identifier` as defined in Section 1 of [@!OpenID.Federation] that is bound to
an entity in a federation. While this Entity Identifier could be any entity in
```

that ecosystem, this entity would usually have the Entity Configuration of a Trust Anchor.
A valid trust path, including the given Entity Identifier, must be constructible from a matching credential.

Below is a non-normative example of such an entry of type `openid_fed`: 
```json
{
  "type": "openid_fed",

>   "values": ["https://trustanchor.example.com"]

Reviewer 11: Since a URL is mentioned, can we add language that prohibits using the URL from retrieving the trust chain because of the privacy implications of such a fetch?

Author: I understand your concern, but I don't think that would make sense in the context of openid federation?

Author: I don't think OpenID4VP would be the right place to do so - profiles like HAIP would be the right place to make recommendations in that regard imho

Author: Hmm maybe it would make sense to add a remark here or in the privacy considerations section that the different mechanisms can have different privacy properties and especially in the case of online lookup, might leak some information?

Building on top of that, HAIP could make choices. Would that work for you?

Reviewer 11: Updated text looks good to me.

the `iss` claim in an SD-JWT) if the mechanisms for `trusted_authorities` are not applicable but might
be less likely to work due to the constraints on value matching (see #selecting_claims for more details).

Each entry in `trusted_authorities` MUST be an object with the following properties:

`type`:
: REQUIRED. A string uniquely identifying the type of information about the issuer trust framework.
Types defined by this specification are listed below.

`values`:
: REQUIRED. An array of strings, where each string (value) contains information specific to the
used Trusted Authorities Query type that allows to identify an issuer, trust framework, or a federation that an
issuer belongs to.

Below are descriptions for the different Type Identifiers (string), the description on how to interpret
and perform the matching logic for each provided value.

Note that depending on the trusted authorities type used, the underlying mechanisms can have
different privacy implications. More detail on privacy considerations for the trusted authorities
can be found in (#privacy_trusted_authorities).

#### Authority Key Identifier

Type:
: `"aki"`

Value:

> : Contains the KeyIdentifier of the AuthorityKeyIdentifier as defined in Section 4.2.1.1 of [@!RFC5280],
> encoded as base64url. The raw byte representation of this element MUST match with the AuthorityKeyIdentifier
> element of an X.509 certificate in the certificate chain present in the credential (e.g., in the header of
> an mdoc or SD-JWT). Note that the chain can consist of a single certificate.

Reviewer 7: ```suggestion
: Contains the KeyIdentifier of the AuthorityKeyIdentifier as defined in Section 4.2.1.1 of [@!RFC5280],
encoded as base64url. The raw byte representation of this element MUST match with the AuthorityKeyIdentifier
element of an X.509 certificate in the certificate chain of the credential signing certificate. Note that the chain can consist of a single certificate, and the credential can include the entire X.509 chain or parts of it.
```

Reviewer 7: I mean the whole chain. For mdocs you have x5chain = doc signer, and the IACA is transmitted out of band. The AKI could also match the IACA, not only the IACA if this makes sense.

Below is a non-normative example of such an entry of type `aki`:
```json
{
  "type": "aki",
  "values": ["s9tIpPmhxdiuNkHMEWNpYim8S8Y"]
}
```

#### ETSI Trusted List

Type:
: `"etsi_tl"`

Value:

> : The identifier of a Trusted List as specified in ETSI TS 119 612 [@ETSI.TL].An ETSI

Reviewer 7: ```suggestion
: The identifier of a Trusted List as specified in ETSI TS 119 612 [@ETSI.TL]. An ETSI
```

In the event that another component is invoked instead of the Wallet, the End-User MUST be informed and give consent before the invoked component returns the `wallet_unavailable` Authorization Error Response to the Verifier.

## Privacy implications of mechanisms to establish trust in issuers {#privacy_trusted_authorities}

> This specification introduces an extension point that allows for a Verifer to express expected issuers or trust frameworks that cerify issuers.

Reviewer 9: ```suggestion
This specification introduces an extension point that allows for a Verifier to express expected issuers or trust frameworks that certify issuers.
```

It is important to understand the implications that different mechanism to establish trust in issuers have on the privacy of the overall system.
Generally speaking, a distinction can be made between self-contained mechanisms, where all information necessary to validate if a credential matches the request is already present in the Wallet and Verifier, and those mechanisms that require some form of online resolution.
Mechanisms that require online resolution can leak information that could be used to profile the usage of credentials and the overall ecosystem.

> Ecosystems that plan to leverage the trusted authorities mechanisms SHOULD make sure that the privacy properties of the mechanisms they choose to support, matches with the desired privcacy properties of the overall ecosystem.

Reviewer 9: ```suggestion
Ecosystems that plan to leverage the trusted authorities mechanisms SHOULD make sure that the privacy properties of the mechanisms they choose to support matches with the desired privacy properties of the overall ecosystem.
```

In the event that another component is invoked instead of the Wallet, the End-User MUST be informed and give consent before the invoked component returns the `wallet_unavailable` Authorization Error Response to the Verifier.

## Privacy implications of mechanisms to establish trust in issuers {#privacy_trusted_authorities}

> This specification introduces an extension point that allows for a Verifier to express expected issuers or trust frameworks that certify issuers.

Reviewer 2: issuers -> Issuers

Author: Fixed in latest commit
